src/core/eme/set_server_certificate.ts

```diff
@@ -68,15 +67,16 @@ function setServerCertificate(
  */
 export default function trySettingServerCertificate(
   mediaKeys : IMockMediaKeys|MediaKeys,
-  serverCertificate : ArrayBuffer|TypedArray,
-  errorStream: Subject<Error|ICustomError>
-) : Observable<null> {
+  serverCertificate : ArrayBuffer|TypedArray
+) : Observable<null|IEMEWarningEvent> {
   return typeof mediaKeys.setServerCertificate === "function" ?
     setServerCertificate(mediaKeys, serverCertificate)
       .pipe(catchError(error => {
         error.fatal = false;
-        errorStream.next(error);
-        return observableOf(null);
+        return observableOf({
+          type: "warning" as "warning",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=305&issues=AWWvglfz_eEZPtUmI_O1&open=AWWvglfz_eEZPtUmI_O1)

[ISSUE_KEY]: AWWvglfz_eEZPtUmI_O1